AWSPinpoint/AWSPinpointAnalyticsClient.m

```diff
@@ -44,13 +44,13 @@ @interface AWSPinpointAnalyticsClient()
 @property (nonatomic, strong) NSMutableDictionary* eventTypeMetrics;
 @property (nonatomic, strong) NSMutableDictionary* globalAttributes;
 @property (nonatomic, strong) NSMutableDictionary* globalMetrics;
-@property (nonatomic, strong) NSDictionary* globalCampaignAttributes;
+@property (nonatomic, strong) NSDictionary* globalFeatureAttributes;
```

Comment:
This is because campaign and journey do not exist at the same time. And the globalCampaignAttributes was used to cache these attributes and attach to events later on. One thing we want to avoid is for one event to have both campaign and journey attributes. This is easier to maintain than caching both and determining which one to attach for following events.

Reply:
Is there a better word than "Feature"?  If i were to look at this code 6 months from now, and try to figure out what a Feature in the context of Pinpoint, I would probably be confused. I'm not fully sold on using this instance variable to serve two purposes but if we do go down this route, i'd probably name this something like globalCampaignOrJourneyAttributes

Reply:
Discussed offline with @wooj2 and @drochetti and agreed upon using this variable. 
Regarding naming, I'm proposing `globalEventSourceAttributes` instead of `globalFeatureAttributes`, since these attributes come from either Campaign events or Journey events.
Regarding [this comment](https://github.com/aws-amplify/aws-sdk-ios/pull/2387#discussion_r401825522), I will expose and explicitly call the `AWSPinpointAnalyticsClient::removeAllGlobalEventSourceAttributes` rather than calling it in `AWSPinpointAnalyticsClient::setEventSourceAttributes`

Reply:
A different proposed name is `globalServerSideAttributes`

Reply:
I like `eventSource` as a generic name for both campaign and journey :)

Reply:
`globalEventSourceAttributes ` it is.